File: spring-cloud-function-demo/src/main/java/com/livk/FunctionApp.java
package com.livk;

import com.livk.common.LivkSpring;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

import java.util.function.Function;

/**
 * <p>
 * FunctionApp
 * </p>
 *
 * @author livk
 * @date 2021/12/3
 */
@SpringBootApplication
public class FunctionApp {
    public static void main(String[] args) {
        LivkSpring.run(FunctionApp.class, args);
    }

    @Bean
    public Function<String, Object> livk() {
        return value -> new StringBuilder(value).reverse().toString();
    }
}

File: spring-cloud-function/src/main/kotlin/com/livk/function/FunctionApp.kt
package com.livk.function

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean

/**
 * @author livk
 */
@SpringBootApplication
open class FunctionApp {
    @Bean
    open fun livk(): (String) -> Any = String::reversed
}

fun main(args: Array<String>) {
    runApplication<FunctionApp>(*args)
}
